lib/natalie/compiler/pass0c.rb

```diff
@@ -20,11 +21,66 @@ def process_iter(exp)
           :iter,
           process_call(call, block_fn=[args,body]),
           args,
-          *body.map {|inst| inst.is_a?(Sexp) ? process(inst) : inst}
+          *body.map { |e| process_atom(e) }
         )
       end
 
+      def process_defined(exp, block_with_args=nil)
+        @in_defined_statement = true
+        exp = exp.new(*exp.map { |e| process_atom(e) })
+        @in_defined_statement = false
+        return exp
+      end
+
+      def type_is_always_truthy(arg)
```

Comment:
could this not be `!type_is_always_falsey` instead?

Reply:
So similar reasoning to the next question
there are types we can say, that they are always truthy or falsey,
but we cannot say the same about variables, call results and the like (yet)
so I just made these helpers with types I know will work fine

Reply:
Yeah, you cannot just do `!type_is_always_falsey` for something like `:lvar`.